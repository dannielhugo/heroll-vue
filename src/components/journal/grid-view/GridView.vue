<template>
  <GridGameList :games="games" />
</template>

<script setup lang="ts">
import type { Game } from '@/models/game';
import GridGameList from './game-list/GridGameList.vue';
defineProps<{ games: Game[] }>();
</script>

<style scoped></style>
