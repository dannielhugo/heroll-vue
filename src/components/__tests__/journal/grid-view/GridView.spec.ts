import { mount } from '@vue/test-utils';
import ElementPlus from 'element-plus';
import GridView from '../../../journal/grid-view/GridView.vue';

describe('GridView', () => {
  test('should mount component', async () => {
    expect(GridView).toBeTruthy();

    const wrapper = mount(GridView, {
      props: {
        games: [],
      },
      global: {
        plugins: [ElementPlus],
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});