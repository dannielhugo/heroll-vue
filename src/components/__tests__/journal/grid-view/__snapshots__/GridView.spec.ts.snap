// Vitest Snapshot v1

exports[`GridView > should mount component 1`] = `"<div class=\\"game-list\\" data-v-ef199a12=\\"\\"></div>"`;
